test/integration/targets/remove_host/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+# test code for the add_host action
+# (c) 2015, Matt Davis <mdavis@ansible.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+- name: add a host to the runtime inventory
+  add_host:
+    name: newdynamichost
+    groups: newdynamicgroup
+    a_var: from add_host
+
+- name: remove host from the runtime inventory
+  remove_host:
+    name: newdynamichost
+
+- name: ensure that dynamically-removed host is not visible via hostvars, groups, etc (there are several caches that could break this)
+  assert:
+    that:
+    - hostvars['newdynamichost'] is not defined
+    - groups['newdynamicgroup'] is defined
+    - "not groups['newdynamicgroup']"
```

Comment:
After this test it would be worth doing another `remove_host` operation, to ensure the action is idempotent.